### ISSUE TITLE
"Crashes" when conditions don't match any rows.

### ISSUE THREAD

**Issue by ryanmjacobs** [harelba/q#114]
For example, if I have a table like this:

| id | age |
| --- | --- |
| 1 | 78 |
| 2 | 45 |
| 3 | 25 |

I get a messy error when trying to query for an age that is _not_ in the table.

``` sql
SELECT * from data.csv where age = 26
```

```
Traceback (most recent call last):
  File "/usr/bin/q", line 1639, in <module>
    run_standalone()
  File "/usr/bin/q", line 1628, in run_standalone
    q_output_printer.print_output(STDOUT,sys.stderr,q_output)
  File "/usr/bin/q", line 1319, in print_output
    self._print_output(f_out,f_err,results)
  File "/usr/bin/q", line 1361, in _print_output
    fmt_str = "%%-%ss" % max_lengths[i]
IndexError: list index out of range
```

Would it be possible to catch this error and not print anything? — like in other SQL tools?

Or even something simple like this?

``` shell
$ q 'SELECT * from data.csv where age = 26'
no rows to display
```

**Comment by harelba** [comment]
Hi @ryanmjacobs - This is supposed to be fixed, and I can't get this error in the latest version 1.5.0.

````
harel@Harels-MBP:~:$ cat file1
id age
1 78
2 45
3 25
harel@Harels-MBP:~:$ q -H "select * from file1 where age = 26"
harel@Harels-MBP:~:$
````

Which version are you using? (````q --version````)

**Comment by ryanmjacobs** [comment]
Hi @harelba, thanks for the quick reply.

I just checked `q --version` and it says that I am on v1.5.0.
However, I just cloned the master branch and ran `bin/q` instead — that worked fine.

**Comment by harelba** [comment]
Sorry for the inconvenience. 

I'll take a look.

**Comment by ernestohs** [comment]
I have the same problem, in my case I don't have a header, the file size is ~9.1 MB and I can't find the "issue" on my data; I did the same as @ryanmjacobs and the version 1.6.0 works for me.

**Comment by harelba** [comment]
hi sorry for the late reply.

@ernestohs thanks for the input - this means that the fix is working well... great :)

**Comment by rsalmei** [comment]
It's the `-HO` that crashes it...
If you run just `q -H "select * from file1 where age = 26"` it doesn't crash indeed, but `q -H "select * from file1 where age = 26"` does.
Version 1.5.0 installed by homebrew. Could you fix it please?

**Comment by harelba** [comment]
@rsalmei Hi, sorry for the late reply, I'm on a long vacation mostly offline.

I'm not sure I can understand the original problem yet. I'm using version 1.5.0 myself on a mac and it doesn't crash even with the `-HO`. Can you please send me "file1" over email so I can check? Please send it zipped so gmail won't modify it or anything.

Thanks
Harel

**Comment by rsalmei** [comment]
Just do:
```
echo 'asd zxc' > x.txt
echo '1 2' >> x.txt
q -HO -b "select * from x.txt where asd=-1"

```
And you get:
```
Traceback (most recent call last):
  File "/usr/local/bin/q", line 1639, in <module>
    run_standalone()
  File "/usr/local/bin/q", line 1628, in run_standalone
    q_output_printer.print_output(STDOUT,sys.stderr,q_output)
  File "/usr/local/bin/q", line 1319, in print_output
    self._print_output(f_out,f_err,results)
  File "/usr/local/bin/q", line 1361, in _print_output
    fmt_str = "%%-%ss" % max_lengths[i]
IndexError: list index out of range
```

**Comment by harelba** [comment]
Fixed in v1.6.3.
